### PR TITLE
feat: Added revision details overview component

### PR DESCRIPTION
**Pull request by pramodcog**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/REV-911
Figma Ref: [Historical 360 Images](https://www.figma.com/file/qUKv93xItnW91t50EuDjyB/3D-Concept-Environment?type=design&node-id=640-106981&t=B9OblzOuRQAcH73k-0)

## Description :pencil:
React component to support 360 Image revision overview and to select them

https://github.com/cognitedata/reveal/assets/87521752/7fee1cea-372b-4649-9925-e3b79639ebdd

## How has this been tested? :mag:
In storybook

## Test instructions :information_source:
1. `cd react-component && yarn build && yarn storybook`
2. Select `Image360HistoricalOverview` from storybook example for `Details` Button
3. Select `Image360HistoricalSummary` from storybook example for `360 Image Revision Details panel`

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx
Comment (larsmoa) [inline react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx:20]: Shouldn't the UI use classes from Reveal that describes the images? And, make revisionDate a `Date`.

File: react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx
Comment (haakonflatval-cognite) [inline react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx:24]: On my system (Firefox, Linux), the scrollbar slightly overlaps with the image list, blocking the date for each revision. Is that fixable?

File: react-components/src/components/utils/Thumbnail.tsx
Comment (haakonflatval-cognite) [inline react-components/src/components/utils/Thumbnail.tsx:21]: Have you tested the toolbar-setup with real data? I imagine it might be a bit hard to do it, but we need to know whether it works :sweat_smile:

File: react-components/src/stories/Image360HistroicalOverviewToolbar.stories.ts
Comment (haakonflatval-cognite) [inline react-components/src/stories/Image360HistroicalOverviewToolbar.stories.ts:5]: Are these comments necessary? I think it's okay to omit links to docs, and assume the reader can find it themselves

File: react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx
Comment (haakonflatval-cognite) [inline react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx:32]: It's fine, I'm just wondering why the `Styled`-prefix is added to some of the variables here? A convention for something?

File: react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx
Comment (haakonflatval-cognite) [inline react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx:43]: I see there is some misalignment in the elements here :sweat_smile: Don't need to fix now, but we should probably have a `prettier` and/or `eslint` configuration for this directory as well

File: react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx
Comment (haakonflatval-cognite) [reply react-components/src/components/Toolbar/Image360HistoricalOverviewToolbar.tsx:24]: https://github.com/cognitedata/reveal/assets/70905152/e2364a00-42d4-403c-8510-49dad90ec876

**Review by haakonflatval-cognite — CHANGES_REQUESTED** [review]
Looks and works nicely! A few comments

**Comment by larsmoa** [comment]
Currently, the Details-component wont automatically trigger this component - correct? Can we set up this so users wont have to do this manually, or hard?

File: react-components/webpack.config.js
Comment (haakonflatval-cognite) [reply react-components/webpack.config.js:35]: I see.. :thinking: Hard to do anything about, I suppose :+1:

File: react-components/src/components/View/Image360HistoricalDetails.tsx
Comment (haakonflatval-cognite) [inline react-components/src/components/View/Image360HistoricalDetails.tsx:61]: What will happen here if `date` or `imageDatas[index]` is undefined? Can we handle those cases properly?

File: react-components/src/components/Toolbar/Image360HistoricalSummary.tsx
Comment (haakonflatval-cognite) [inline react-components/src/components/Toolbar/Image360HistoricalSummary.tsx:22]: Same as with `Image360HistoricalDetails` - do we need this many arguments, can we do with just the 360 image entity?

File: react-components/src/components/View/Image360HistoricalDetails.tsx
Comment (haakonflatval-cognite) [inline react-components/src/components/View/Image360HistoricalDetails.tsx:114]: From the design, it looks like we would want the detail summary to stretch along the entire bottom part of the Reveal Viewer. But that is probably more relevant to the related Data Explorer PR.

**Review by haakonflatval-cognite — APPROVED** [review]
Looks and feels great! Pointed out some suggestions

File: react-components/src/components/View/Image360HistoricalDetails.tsx
Comment (pramodcog) [reply react-components/src/components/View/Image360HistoricalDetails.tsx:18]: Earlier entity did not exposed these data, have updated them now :)

File: react-components/src/components/View/Image360HistoricalDetails.tsx
Comment (pramodcog) [reply react-components/src/components/View/Image360HistoricalDetails.tsx:61]: It is handle, you can check undefined case in the storybook

Comment (haakonflatval-cognite) [reply react-components/src/components/View/Image360HistoricalDetails.tsx:48]: We concluded that it's still unhandled

**Review by haakonflatval-cognite — APPROVED** [review]
I think it's getting good now :+1: LGTM when the rest of the comments are addressed

**Pull request merged**
